### PR TITLE
Added a new convenience Path class to define path from center, length, and position angle

### PR DESCRIPTION
**Pull request by astrofrog**
This includes an option to sample the path using a specific number of points for when the field of view becomes large.

This still needs tests and docs, but @keflavich - does this look ok to you otherwise?

### CODE REVIEW

File: pvextractor/geometry/helpers.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        two end points. For small fields of view, this will be a good
+        approximation to the path, but for larger fields of view, where
+        spherical distortions become important, this should be increased to
+        provide a smooth path.
+    width : None or float or :class:`~astropy.units.Quantity`
+        The width of the path. If ``coords`` is passed as a list of pixel
+        positions, the width should be given (if passed) as a floating-point
+        value in pixels. If ``coords`` is a coordinate object, the width
+        should be passed as a :class:`~astropy.units.Quantity` instance with
+        units of angle.
```
Comment (keflavich) [inline pvextractor/geometry/helpers.py]: Should point out the behavior of `None` here - presumably, `None` means simple interpolation?

File: pvextractor/geometry/helpers.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        # Convert the central position to cartesian coordinates
+        c1, c2, c3 = center.cartesian.xyz.value
+
+        # Find the end points of the path
+        clon, clat = center.spherical.lon, center.spherical.lat
+        plat = clat + np.linspace(length * 0.5, -length * 0.5, sample)
+        x, y, z = UnitSphericalRepresentation(clon, plat).to_cartesian().xyz.value
+
+        print(x)
```
Comment (keflavich) [inline pvextractor/geometry/helpers.py]: is this a debug statement?

File: pvextractor/geometry/helpers.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        The width of the path. If ``coords`` is passed as a list of pixel
+        positions, the width should be given (if passed) as a floating-point
+        value in pixels. If ``coords`` is a coordinate object, the width
+        should be passed as a :class:`~astropy.units.Quantity` instance with
+        units of angle.
+
+    Notes
+    -----
+    The orientation of the final path will be such that for a position angle of
+    zero, the path goes from North to South. For a position angle of 90
```
Comment (keflavich) [inline pvextractor/geometry/helpers.py]: This seems like a strange convention.  I'm used to zero either pointing east->west or south->north.  Isn't position angle defined with 0 = east->west?

File: pvextractor/geometry/helpers.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        # z' = z cos a + (1 - cos a)(c3c1x + c3c2y + c3c3z) + (c1y - c2x)sin a
+        #
+        # Source: https://www.uwgb.edu/dutchs/MATHALGO/sphere0.htm
+
+        cosa = np.cos(angle)
+        sina = np.sin(angle)
+
+        xd = x * cosa + (1 - cosa) * (c1*c1*x + c1*c2*y + c1*c3*z) + (c2 * z - c3 * y) * sina
+        yd = y * cosa + (1 - cosa) * (c2*c1*x + c2*c2*y + c2*c3*z) + (c3 * x - c1 * z) * sina
+        zd = z * cosa + (1 - cosa) * (c3*c1*x + c3*c2*y + c3*c3*z) + (c1 * y - c2 * x) * sina
```
Comment (keflavich) [inline pvextractor/geometry/helpers.py]: I'm just going to trust that this is right, not check it in detail.  Transcription from the comment above looks OK.

**Comment by keflavich** [comment]
:+1: 
I hope this means you're making an interface in Glue that will allow you to rotate a pvextraction around?  e.g., as in
https://www.dropbox.com/s/j1jp2o3htddo16f/rotation_movie.mp4?dl=0

File: pvextractor/geometry/helpers.py (hunk 1 above)
Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: This was copied and pasted from ``None``, so I'll fix both.

File: pvextractor/geometry/helpers.py (hunk 2 above)
Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: Oops

File: pvextractor/geometry/helpers.py (hunk 3 above)
Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: In astronomy, position angles are defined from North, no?

Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: https://en.wikipedia.org/wiki/Position_angle

Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: I could swap it so that it goes south -> north instead of north->south though

Comment (keflavich) [inline pvextractor/geometry/helpers.py]: Right.... that's... why I always get confused about this.  But aren't you defining them from South, in this case?  I thought a vector pointing north would be zero even in that definition?  the wikipedia article unfortunately did not make things clear for me.

Comment (astrofrog) [inline pvextractor/geometry/helpers.py]: Oh, the direction of the PV slice is arbitrary to some extent, but indeed I can flip it around - not an issue :)

**Comment by astrofrog** [comment]
Will work on docs and tests this afternoon

**Comment by keflavich** [comment]
ping me if you want further review on that

**Comment by astrofrog** [comment]
@keflavich - ready for final review. Note that this includes a couple other minor fixes.

**Comment by keflavich** [comment]
Looks fine.  Somewhere in the unspecified distant future, we might want to add an example showing how to display this class as an arrow or vector in mpl

**Comment by astrofrog** [comment]
Yes, I think it would be nice to have more plotting examples in general.

**Comment by astrofrog** [comment]
Hmm at the moment the PV diagram is not exactly centered on that position, need to look into it a bit more.

**Comment by astrofrog** [comment]
I fixed the offset issues in #68 and #69. If tests pass, this is ready to merge.

**Pull request merged**